CMakeLists.txt

```
@@ -105,6 +105,7 @@ function(trusted_firmware_build)
                ${TFM_REGRESSION_ARG}
                ${TFM_PROFILE_ARG}
                -DTFM_TEST_REPO_PATH=${ZEPHYR_TFM_MODULE_DIR}/tf-m-tests
+               -DMCUBOOT_PATH=${ZEPHYR_TFM_MODULE_DIR}/mcuboot
```

Why is it not possible to use:
```suggestion
               -DMCUBOOT_PATH=${ZEPHYR_MCUBOOT_MODULE_DIR}
```

and thus use the same mcuboot as Zephyr ?
And then completely remove mcuboot from this repository.

Are they not compatible ?

---

They are, and when I tried it it works, they just aren't identical I think. So I think to keep TFM exactly as released, we don't want to do that yet. Though I'm not entirely convinced myself why we can't.

---

I understand that reason, but in that case, why not use `west` ?
That would remove the need to upmerge mcuboot in two places:
- This repo
- https://github.com/zephyrproject-rtos/mcuboot

Just add this to `west.yml`:

```
    - name: tfm_mcuboot
      repo-path: mcuboot
      path: modules/tee/tfm/mcuboot
      revision: 69344636bee1f18a6703dc5a8e7a0edfd8128d45
#      revision: 1.7.0-rc1
```
Note, the `69344636bee1f18a6703dc5a8e7a0edfd8128d45` SHA is part of this PR: https://github.com/zephyrproject-rtos/mcuboot/pull/39
https://github.com/zephyrproject-rtos/mcuboot/pull/39/commits/69344636bee1f18a6703dc5a8e7a0edfd8128d45

But if you ask @nvlsianpu, then maybe he can include the mcuboot tags in the upmerge, and you will be able to use `1.7.0-rc1` instead of the SHA.

That way we remove the need to upmerge mcuboot inside this repo, but still use a different version for TFM, than what Zephyr is using.
And there is only one place to keep mcuboot in sync with upstream.

---

> But if you ask @nvlsianpu, then maybe he can include the mcuboot tags in the upmerge, and you will be able to use 1.7.0-rc1 instead of the SHA

I can send 1.7.0-rc1 tag to zephyr's mcuboot fork If needed.

---

Changed to use west now. See if you like it better

---

I'm OK with this approach, but we should use the exact MCUBoot commit they are using in upstream TF-M at the point that we update the Zephyr TF-M module here, just to avoid any subtle issues or complications.

---

@microbuilder that would be the optimal, and if this guide is followed:
https://docs.zephyrproject.org/latest/guides/modules.html#allowed-practices
>  Upstream changes brought in by performing a Git merge of the intended upstream branch (e.g. main branch, latest release branch, etc.) submitting the result in pull request against the module main branch, and merging the pull request using the Create a merge commit operation. This approach is applicable to modules with an upstream project Git repository. The main advantages of this approach is that the upstream repository history (that is, the original commit SHAs) is preserved in the module repository. 

then the upstream SHAs should be available.

---

> Changed to use west now. See if you like it better

You need a commit with:
```
git rm -r mcuboot
```
for this to work.

---

> @microbuilder that would be the optimal, and if this guide is followed:

Just verified, https://github.com/zephyrproject-rtos/mcuboot/pull/39, does have all the upstream SHAs.

---

Should I change the west entry to a sha, or should we bring in the tag?

---

> You need a commit with:
> 
> ```
> git rm -r mcuboot
> ```
> 
> for this to work.

Done

---

> > But if you ask @nvlsianpu, then maybe he can include the mcuboot tags in the upmerge, and you will be able to use 1.7.0-rc1 instead of the SHA
> 
> I can send 1.7.0-rc1 tag to zephyr's mcuboot fork If needed.

@nvlsianpu that would be highly appreciated :smile: